src/app/features/courses/components/courses-edit/courses-edit.component.ts

```diff
@@ -26,55 +26,82 @@ export class CoursesEditComponent {
 
   public courseForm!: FormGroup;
 
+  private subs: Subscription[] = [];
+
   constructor(
     private coursesService: CoursesService,
     private router: Router,
     private route: ActivatedRoute,
     private fb: FormBuilder
   ) {
     const courseId = this.route.snapshot.params['id'];
-    this.courseForm = this.createForm(courseId);
+    this.createForm(courseId);
   }
 
   onSave() {
     if (this.operationType === 'edit') {
-      this.coursesService.updateCourse(this.courseForm.value);
+      this.subs.push(
+        this.coursesService
+          .updateCourse(this.courseForm.value)
+          .subscribe(() => {
+            this.router.navigate(['/courses']);
+          })
+      );
     } else {
-      this.coursesService.createCourse(this.courseForm.value);
+      this.subs.push(
+        this.coursesService
```

Comment:
All HTTP calls resolve with fail or success and then observable immediately completes, no need to unsubscribe here. Please, remove unsubscription for each HTTP call method you use.